src/main/cc/endpoints/fpga_memory_model.h

```diff
@@ -14,6 +14,30 @@
 constexpr int HISTOGRAM_SIZE = 1024;
 constexpr int BIN_SIZE = 36;
 constexpr data_t BIN_H_MASK = (1L << (BIN_SIZE - 32)) - 1;
+constexpr data_t NUM_RANGE_MASK = (1L << 4) - 1;
+constexpr data_t RANGE_BYTES_MASK = (1L << 16) - 1;
+constexpr uint64_t MEM_SIZE (1L << 35);
```

Comment:
This seems brittle. Consider passing or reading this out.

Reply:
Then should we do the same for BIN_H_MASK?

Reply:
I was mostly referring to the memory system size, but yeah, that's a good idea. I hard coded 36 since that's the width of the BRAM port, but given that we expect to change the bin sizes more frequently it should be passed out too.